### ISSUE TITLE
Add functions to work with AIMS data?

### ISSUE THREAD

**Issue by dbarneche** [open-AIMS/ereefs#33]
Would it be possible to also work with AIMS [eReefs data service](https://hyrax.ereefs.aims.gov.au/)?
This is probably a high priority function as this might affect a lot the inner working of multiple functions.
If so we could create a function for this, or expand capacity of all functions.

**Comment by BarbaraRobson** [comment]
Yes. Two ways to do this.  The better way would be to use an alternative (and simpler) plotting function if it is AIMS data service data. The alternative would be to use the same plotting functions but calculate the grid cell corners for the AIMS regraded data, and save that in sysdata.rda


[1520817071662_PastedImage]

________________________________
From: Diego Barneche ***@***.***>
Sent: 23 November 2022 13:56
To: open-AIMS/ereefs ***@***.***>
Cc: Barbara Robson ***@***.***>; Assign ***@***.***>
Subject: [open-AIMS/ereefs] Add functions to work with AIMS data? (Issue #33)

CAUTION: This email originated from outside of AIMS.
DO NOT click links or open attachments in unsolicited emails, call and confirm the content is safe with the sender first. To report any suspicious emails, use the Report Message outlook add-in or send to ***@***.***


Would it be possible to also work with AIMS eReefs data service<https://hyrax.ereefs.aims.gov.au/>?
This is probably a high priority function as this might affect a lot the inner working of multiple functions.
If so we could create a function for this, or expand capacity of all functions.

**Comment by dbarneche** [comment]
Would it be possible to split the internal functionality of, e.g. `map_ereefs` into:
- standardising data
- plotting data
The above would then be handled by two internal functions, with the one being able to handle both NCI and AIMS data such that it puts them into the same format for the second internal plotting function? Does that make sense / is it feasible?
Sorry that might be too naive a question given I don't know much about the how the different datasets compare or differ.

**Comment by BarbaraRobson** [comment]
If we take my second suggested approach (calculating and storing the locations of grid cell corners for AIMS data) then yes, we could do it that way.

The original ereefs model data (and that on the NCI server) is on an irregular (curvilinear) grid. We have the locations of the cell centres but also need the locations of the cell corners to plot it accurately. When plotted on the scale of the whole GBR, it doesn't matter much and we could just use the cell centre locations and could probably use ggplot2::geom_contour_filled() to plot it, but when zooming in to small areas of coastal or near-island waters, this can be inaccurate. Therefore, I have instead used ggplot2::geom_polygon() to plot the data after first setting up a polygon for every grid cell in the model domain. This is quite slow, but accurate around the edges.

The AIMS data centre version has regridded the original model output onto a regular grid (evenly spaced, rectangular grid cells). It won't be quite as accurate in the edge cases that I am concerned about above, but is much easier to handle. We can simply use ggplot2::geom_tile() to map it (or alternatively, could use geom_contour_filled() or even the functions that come with the raster package. If we do it this way, it will be much faster, especially plotting gifs for animations. (Alternatively, we could treat it in the same way as the NCI-served data after first calculating the locations of the grid corners, but doing it that way is slower for no advantage).

I think using a different plotting function for the AIMS regridded data would be better -- faster and cleaner.

Barbara

________________________________
